### ISSUE TITLE
Incorrect return for getAuth response then the expected type

### ISSUE THREAD

**Issue by vishalj** [agencyenterprise/react-native-health#42]
**Describe the bug**
Incorrect types and enum

**To Reproduce**
1. Eslint error for (the = in `HealthStatusCode = {` is not needed)
```
export enum HealthStatusCode = {
    NotDetermined = 0,
    SharingDenied = 1,
    SharingAuthorized = 2,
  }
```

2. Steps to reproduce the behavior:
const permissions = {
 permissions: {
    read: [
      AppleHealthKit.Constants.Permissions.Workout,
      AppleHealthKit.Constants.Permissions.ActiveEnergyBurned,
      AppleHealthKit.Constants.Permissions.BasalEnergyBurned,
      AppleHealthKit.Constants.Permissions.HeartRate,
    ],
    write: [
      AppleHealthKit.Constants.Permissions.Workout,
      AppleHealthKit.Constants.Permissions.ActiveEnergyBurned,
      AppleHealthKit.Constants.Permissions.BasalEnergyBurned,
      AppleHealthKit.Constants.Permissions.HeartRate,
    ],
} as HealthKitPermissions

AppleHealthKit.getAuthStatus(permissions, (err: string, results: HealthStatusResult) => {
  console.log(results)
})
following is the console log
```
{permissions: {…}}
permissions:
read: (4) [0, 0, 0, 0]
write: (4) [0, 0, 0, 0]
__proto__: Object
__proto__: Object
```
Results contain `permissions` property that is not defined in the 
``` 
export interface HealthStatusResult {
    read: HealthStatusCode[],
    write: HealthStatusCode[],
  }
``` 
  And as a result when using results.write, it is returned as undefined and if I use results.permissions.write then eslint shows error as permissions is not defined on results.

**Expected behavior**
The eslint should not throw error. And the return should match to the defined type

The return type matches the expected type

**Screenshots**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by NS-BOBBY-C** [comment]
I recently fixed this in #39. I'll fix the enum part right away, but I wonder if there is more that needs to change with the implementation of the getAuthStatus native function for iOS

**Comment by vishalj** [comment]
> I recently fixed this in #39. I'll fix the enum part right away, but I wonder if there is more that needs to change with the implementation of the getAuthStatus native function for iOS

I actually am using the new code you added in #39, in the return type the results contains permissions property that is not expected in the `HealthStatusResult` type which is causing the eslint to fail

**Comment by NS-BOBBY-C** [comment]
Gotcha. I'll work on a fix right now open a PR.

**Comment by NS-BOBBY-C** [comment]
Sorry about that! Just to clarify - it's just the typing not working with the linter? I've opened a PR (#43)

```typescript
  export enum HealthStatusCode {
    NotDetermined = 0,
    SharingDenied = 1,
    SharingAuthorized = 2,
  }

  export interface HealthStatusResult {
    permissions: {
      read: HealthStatusCode[],
      write: HealthStatusCode[],
    }
  }
```

**Comment by vishalj** [comment]
I think that should work nicely now. Thanks.

**Comment by lucaspbordignon** [comment]
Hey @vishalj!

Thanks a lot for submitting that request. I just merged PR #43 from @NS-BOBBY-C and will be releasing a new minor version with the fix!
